types/passport-twitch-latest/index.d.ts

```diff
@@ -0,0 +1,85 @@
+// Type definitions for passport-twitch-latest 1.0
+// Project: https://github.com/sascha-beloborodov/passport-twitch
+// Definitions by: Charlie Laabs <https://github.com/claabs>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.5
+
+import * as PassportOauth2 from 'passport-oauth2';
+import * as passport from 'passport';
+import * as e from 'express';
+
+export class Strategy extends PassportOauth2 {
+    constructor(options: StrategyOptions, verify: VerifyFunction);
+    constructor(options: StrategyOptionsWithRequest, verify: VerifyFunctionWithRequest);
+
+    userProfile(accessToken: string, done: (err?: Error | null, profile?: TwitchProfile) => void): void;
+    authorize(strategy: string | string[], callback?: (...args: any[]) => any): AuthenticateRet;
+    authorize(
+        strategy: string | string[],
+        options: AuthenticateOptions,
+        callback?: (...args: any[]) => any,
+    ): AuthenticateRet;
+    authenticate(req: e.Request, options?: AuthenticateOptions): void;
+}
+
+export class OAuth2Strategy extends Strategy {}
+
+export type Optional<T, K extends keyof T> = Omit<T, K> & Partial<T>;
+
+export type StrategyOptions = Optional<PassportOauth2.StrategyOptions, 'authorizationURL' | 'tokenURL'>;
+export type StrategyOptionsWithRequest = Optional<
+    PassportOauth2.StrategyOptionsWithRequest,
+    'authorizationURL' | 'tokenURL'
+>;
+
+export type VerifyFunction =
+    | ((
+          accessToken: string,
+          refreshToken: string,
+          profile: TwitchProfile,
+          verified: PassportOauth2.VerifyCallback,
+      ) => void)
```

Comment:
Please move this to the bottom of the list, otherwise it wont' be taken and the sample code from the README wont' compile:
```ts
(accessToken, refreshToken, profile, done) => {
```
Less specific definition should be the latest:
https://www.typescriptlang.org/docs/handbook/functions.html#overloads

Reply:
Just changing the order of the functions didn't make the README sample compile, since Typescript infers the longest callback parameter by default, and the final callback parameter is a required callback. 

The solution I put in place makes inference on the sample code work by default, but requires manual casting on the longer overload at the cost of some tslint disables.